uaclient/entitlements/repo.py

```diff
@@ -213,19 +218,29 @@ def setup_apt_config(self):
             try:
                 util.subp(
                     ['apt-get', 'install', '--assume-yes'] + prerequisite_pkgs,
-                    capture=True, retry_sleeps=APT_RETRIES)
+                    retry_sleeps=apt.APT_RETRIES)
             except util.ProcessExecutionError as e:
-                logging.error(str(e))
+                message = 'APT install failed.'
+                if 'Could not get lock /var/lib/dpkg/lock' in str(e.stderr):
+                    message += ' Another process is running APT'
+                logging.error(message)
                 return False
         apt.add_auth_apt_repo(repo_filename, repo_url, token, repo_suites,
                               keyring_file)
         # Run apt-update on any repo-entitlement enable because the machine
         # probably wants access to the repo that was just enabled.
-        # Side-effect is that apt policy will new report the repo as accessible
+        # Side-effect is that apt policy will now report the repo as accessible
         # which allows ua status to report correct info
         print('Updating package lists')
-        util.subp(
-            ['apt-get', 'update'], capture=True, retry_sleeps=APT_RETRIES)
+        try:
+            util.subp(
+                ['apt-get', 'update'], retry_sleeps=apt.APT_RETRIES)
+        except util.ProcessExecutionError as e:
+            message = 'APT update failed.'
+            if 'Could not get lock /var/lib/dpkg/lock' in str(e.stderr):
+                message += ' Another process is running APT'
+            logging.error(message)
```

Comment:
Should we be using a `UserFacingError` here?

Reply:
I would say no here because we also want a followup user facing error "cannot enable service X'. I guess we could collect and report both messages